OpenRA.Mods.Common/Pathfinder/HierarchicalPathFinder.cs

```diff
@@ -629,9 +629,18 @@ static Grid GetGrid(CPos cellInGrid, Grid mapBounds)
 				var sourcesWithPathableNodes = new List<CPos>(sourcesWithReachableNodes.Count);
 				foreach (var source in sourcesWithReachableNodes)
 				{
-					reverseAbstractSearch.TargetPredicate = cell => cell == source;
-					if (reverseAbstractSearch.ExpandToTarget())
-						sourcesWithPathableNodes.Add(source);
+					var sourceStatus = reverseAbstractSearch.Graph[source];
+					if (sourceStatus.Status == CellStatus.Closed)
```

Comment:
A short comment here that a previous search might have expanded the source already would be good.

Reply:
Updated